dialog/src/main/java/roboy/dialog/states/botboy/BotBoyIntroductionState.java

```diff
@@ -0,0 +1,214 @@
+        }
+
+
+        // 2. get interlocutor object from context
+        // this also should query memory and do other magic
+        Interlocutor person = getContext().ACTIVE_INTERLOCUTOR.getValue();
+        person.addName(name);
+        // Roboy roboy = new Roboy(getMemory());
+
+
+        // 3. update interlocutor in context
+        updateInterlocutorInContext(person);
+        String retrievedPersonalFact = "";
+        Double segueProbability = 0.0;
+
+        // 4. check if person is known/familiar
+        if (person.FAMILIAR) {
+            // 4a. person known/familiar
+            RandomList<MemoryNodeModel> nodes = getMemNodesByIds(person.getRelationships(FRIEND_OF));
+            if (!nodes.isEmpty()) {
+                retrievedPersonalFact = " You are friends with " +
+                        nodes.getRandomElement().getProperties().get(Neo4jProperty.name).toString();
+            }
+
+            RelationshipAvailability availability = person.checkRelationshipAvailability(personPredicates);
+            if (availability == SOME_AVAILABLE) {
+                nextState = (Math.random() < 0.3) ? getTransition(UPDATE_KNOWN_PERSON) : getTransition(LEARN_ABOUT_PERSON);
+            } else if (availability == NONE_AVAILABLE) {
+                nextState = getTransition(LEARN_ABOUT_PERSON);
+            } else {
+                nextState = getTransition(UPDATE_KNOWN_PERSON);
+            }
+        } else {
+            // 4b. person is not known
+            nextState = getTransition(LEARN_ABOUT_PERSON);
+            segueProbability = 0.6;
+        }
+        String retrievedRoboyFacts = getRoboyFactsPhrase(new Roboy(getMemory()));
+        Segue s = new Segue(Segue.SegueType.DISTRACT, segueProbability);
+
+        if(person.FAMILIAR) {
+            return Output.say(getResponsePhrase(person.getName(), person.FAMILIAR) +
+                    retrievedPersonalFact + " It is great to chat with you again!").setSegue(s);
+        }
+
+        return Output.say(getResponsePhrase(person.getName(), person.FAMILIAR) +
+                retrievedPersonalFact + retrievedRoboyFacts).setSegue(s);
+    }
+
+    @Override
+    public State getNextState() {
+        return nextState;
+    }
+
+    private String getNameFromInput(Interpretation input) {
+        return getInference().inferProperty(name, input);
+    }
+
+    private void updateInterlocutorInContext(Interlocutor interlocutor) {
+        getContext().ACTIVE_INTERLOCUTOR_UPDATER.updateValue(interlocutor);
+    }
+
+    private String getIntroPhrase() {
+        return introPhrases.getRandomElement();
+    }
+
+    private String getResponsePhrase(String name, boolean familiar) {
+        if (familiar) {
+            return String.format(successResponsePhrases.getRandomElement(), name);
+        } else {
+            return String.format(failureResponsePhrases.getRandomElement(), name);
+        }
+    }
+
+    private String getRoboyFactsPhrase(Roboy roboy) {
```

Comment:
@SezerCan

Reply:
I do not know why but it does not always give the true result. Roboy is always 4 month old. I am not sure if it is because of my system or not. So are you sure? @missxa

Reply:
for me this works fine. i'll let it go through